samples/bluetooth/peripheral_fast_pair/src/main.c

```diff
@@ -162,6 +162,17 @@ BT_CONN_CB_DEFINE(conn_callbacks) = {
 	.security_changed = security_changed,
 };
 
+static void pairing_complete(struct bt_conn *conn, bool bonded)
+{
+	if (bonded) {
+		fp_adv_mode = BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_SHOW_UI_IND;
+		k_work_reschedule(&fp_adv_mode_status_led_handle, K_NO_WAIT);
+		if (!peer) {
+			advertising_start();
+		}
```

Comment:
I think we always must have a peer here

Reply:
What if peer immediately disconnects? Can this function be interrupted nd the disconnected callback be executed?

Reply:
This cannot happen, because we assume that we use cooperative contexts for Bluetooth threads and system workq

Reply:
btw. If we could be preempted by disconnected callback in sample's code, we could also encounter various other issues ;)